crates/huub/src/constraints/int_diffn.rs

```diff
@@ -0,0 +1,829 @@
+						curr_obj_idx,
+						d,
+						self.lb_tracker[curr_obj_idx][d],
+						self.ub_tracker[curr_obj_idx][d],
+						self.ub_tracker[curr_obj_idx][d] + 1
+					);
+					reason.push(actions.get_int_lit(
+						self.box_posn[curr_obj_idx][d],
+						IntLitMeaning::Less(self.ub_tracker[curr_obj_idx][d] + 1),
+					));
+				} else {
+					trace!(
+						"reason of obj {:?} in dimension {} [{:?}..{:?}] >= {}",
+						curr_obj_idx,
+						d,
+						self.lb_tracker[curr_obj_idx][d],
+						self.ub_tracker[curr_obj_idx][d],
+						self.lb_tracker[curr_obj_idx][d]
+					);
+					reason.push(actions.get_int_lit(
+						self.box_posn[curr_obj_idx][d],
+						IntLitMeaning::GreaterEq(self.lb_tracker[curr_obj_idx][d]),
+					));
+				}
+			} else {
+				trace!(
+					"reason of obj {:?} in dimension {} [{:?}..{:?}] < {}",
+					curr_obj_idx,
+					d,
+					self.lb_tracker[curr_obj_idx][d],
+					self.ub_tracker[curr_obj_idx][d],
+					self.ub_tracker[curr_obj_idx][d] + 1
+				);
+				reason.push(actions.get_int_lit(
+					self.box_posn[curr_obj_idx][d],
+					IntLitMeaning::Less(self.ub_tracker[curr_obj_idx][d] + 1),
+				));
+				trace!(
+					"reason of obj {:?} in dimension {} [{:?}..{:?}] >= {}",
+					curr_obj_idx,
+					d,
+					self.lb_tracker[curr_obj_idx][d],
+					self.ub_tracker[curr_obj_idx][d],
+					self.lb_tracker[curr_obj_idx][d]
+				);
+				reason.push(actions.get_int_lit(
+					self.box_posn[curr_obj_idx][d],
+					IntLitMeaning::GreaterEq(self.lb_tracker[curr_obj_idx][d]),
+				));
+			}
+		}
+		reason.extend(self.explain_fr(actions, all_fr, fr_support, curr_obj_idx));
+		reason
+	}
+}
+
+impl<P, E> Propagator<P, E> for IntDiffnSweep
+where
+	P: PropagationActions,
+	E: ExplanationActions,
+{
+	#[tracing::instrument(name = "diffn", level = "trace", skip(self, actions))]
+	fn propagate(&mut self, actions: &mut P) -> Result<(), Conflict> {
+		for o in 0..self.box_posn.len() {
+			for d in 0..self.dimensions {
+				self.ub_tracker[o][d] = actions.get_int_upper_bound(self.box_posn[o][d]);
+				self.lb_tracker[o][d] = actions.get_int_lower_bound(self.box_posn[o][d]);
+				self.lb_sizes[o][d] = actions.get_int_lower_bound(self.box_size[o][d]);
+			}
+		}
+
+		for o_idx in 0..self.box_posn.len() {
+			// Skip if target property has been lost
+			if actions.get_trailed_int(self.target[o_idx]) == 1 {
+				continue;
+			}
+
+			if self.non_strict
+				&& (0..self.dimensions)
+					.any(|d| actions.get_int_val(self.box_size[o_idx][d]) == Some(0))
+			{
+				continue;
+			}
+			let mut fr_support: Vec<usize> = Vec::new();
+
+			if let Some((all_fr, all_fr_explain)) =
+				self.generate_fr::<P>(actions, &mut fr_support, o_idx, self.dimensions)
+			{
+				if self.fixed_in_all_dimensions(o_idx) {
+                    trace!("CONFLICT");
+					// Conflict will occur here since there exists forbidden regions for a fixed
+					// object
+					// TODO: Conflict occurs in all dimensions but we only reason about it in one
```

Comment:
You are right. You can point at each dimension as cause of the conflict. However, once you resolve one (allow movement in one dimension) then no more conflict exists.

Potentially, this means we could output all conflict clauses or be smart in some other way, but the present strategy of picking the first dimension seems good to me. We can probably remove the TODO.